lib/SimpleSAML/Module.php

```diff
@@ -126,31 +149,20 @@ public static function process(Request $request = null)
             $request = Request::createFromGlobals();
         }
 
-        if ($request->getPathInfo() === '/') {
-            throw new Error\NotFound('No PATH_INFO to module.php');
-        }
-
-        $url = $request->getPathInfo();
-        assert(substr($url, 0, 1) === '/');
+        $basepathinfo = $request->getPathInfo();
+        $rewpathinfo = $_SERVER['PATH_INFO'];
 
-        /* clear the PATH_INFO option, so that a script can detect whether it is called with anything following the
-         *'.php'-ending.
-         */
-        unset($_SERVER['PATH_INFO']);
-
-        $modEnd = strpos($url, '/', 1);
-        if ($modEnd === false) {
-            // the path must always be on the form /module/
-            throw new Error\NotFound('The URL must at least contain a module name followed by a slash.');
+        if ($basepathinfo === '/') {
+            throw new Error\NotFound('No PATH_INFO to module.php');
```

Comment:
This error would be shown to the end user, but won't help much, I'm afraid. Should we have an error message similar to the old one?

Reply:
I agree that the error message is not giving much valuable information, so using the old one is a good option.

About the use of  $_SERVER['path_info'] , i didn't find a way to have it through Symfony request. i had a look at the code inside symfony and the thing is they are trying to be "too smart" about request parsing, providing a "too high" abstraction level above the initial request. so loosing information with no way to get it back from the original request. So the only way i found was to ask the original request details directly through $_SERVER.

Reply:
Uhm, I see...

Do you think this is actually wrong behaviour in Symfony? Should we report it upstream for them to consider a change (regardless of whether we work around it or not)?

Reply:
I think this is a behaviour that is not ideal for the purpose we want to achieve (ie, detecting which module is targeted by the  rewritten url) , i cannot say it's a "bad" behaviour from symfony, it's just a choice that was made that is making the request object not well suited for low level url access but very interesting to process request content & structure.